javalib/src/main/scala/java/util/ArrayDeque.scala

```diff
@@ -0,0 +1,241 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+// Contains original work for ScalaNative which falls under the Scala Native
+// license.
+
+package java.util
+
+/// ScalaNative Porting Note:
+///
+///     * Ported, with thanks & gratitude, from ScalaJS ArrayDeque.scala
+///       commit 9DC4D5b, dated 2018-10-12.
+///
+///     * ScalaNative ArrayList is the inner type, rather than js.Array.
+///
+///     * equals() & hashcode() are not implemented so comparing ArrayDeque
+///       instances ad1 == ad2 is bit of a pain. It gives object equality
+///       not the more useful contents equality. Someday...
+///           https://alvinalexander.com/scala/ \
+///           how-to-define-equals-hashcode-methods-in-scala-object-equality
```

Comment:
`equals` and `hashCode` are inherited from `AbstractList`, which has a correct implementation. This comment is misleading and should be removed.

Reply:
Done, thank you for the insight & correction.